### ISSUE TITLE
A strange lambda in the FreeApplicative example

### ISSUE THREAD

**Issue by denisftw** [typelevel/cats#1471]
Some people pointed out that there is a strange lambda in the example of [FreeApplicative](http://typelevel.org/cats/datatypes/freeapplicative.html):

```scala
val compiler =
   λ[FunctionK[ValidationOp, FromString]] { fa =>
      str =>
        fa match {
          case Size(size) => str.size >= size
          case HasNumber  => str.exists(c => "0123456789".contains(c))
        }
   }
```

Not sure what it does here, but it doesn't compile. Shouldn't it be:

```scala
val compiler = new FunctionK[ValidationOp, FromString] {
      def apply[A](fa: ValidationOp[A]): FromString[A] = str =>
          fa match {
            case Size(size) => str.size >= size
            case HasNumber  => str.exists(c => "0123456789".contains(c))
          }
      }
```

A similar thing happens with other compilers.

**Comment by peterneyens** [comment]
It uses the [kind projector](https://github.com/non/kind-projector) compiler plugin (more specifically its syntax for [polymorphic lambas](https://github.com/non/kind-projector/#polymorphic-lambda-values)).

It is good that you raise the issue, it was mentioned before that we should probably use the regular syntax in the documentation. We would love a PR with these changes, if you are interested.

**Comment by denisftw** [comment]
Ah, right. For some reason, I though that kind projector only uses the question marks syntax.

Thanks!

**Comment by peterneyens** [comment]
This seems to be a copy of #1449.

**Comment by denisftw** [comment]
I suppose it's different. Rob is referring to the `~>` thing, which is just an alternative syntax for `FunctionK`. If I remember correctly, this type had alias `NaturalTranformation` several months ago, but then it was changed to `FunctionK`. My changes don't affect this, because `FunctionK` was already there.

**Comment by ceedubs** [comment]
Resolved by #1472
